tests/testthat/test-listings.R

```diff
@@ -247,3 +247,46 @@ testthat::test_that("unique_rows removes duplicate rows from listing", {
   )
   expect_equal(expected_strings, result_strings)
 })
+
+testthat::test_that("as_listing custom format works in key cols", {
+  lsting <- as_listing(
+    ex_adsl[1:10, ],
+    key_cols = c("AGE", "BMRKR1"),
+    disp_cols = c("SEX", "ARM"),
+    default_formatting = list(all = fmt_config(), numeric = fmt_config(format = "xx.xx"))
+  )
+
+  testthat::expect_identical(matrix_form(lsting)$strings[2, 1:2], c(AGE = "24.00", BMRKR1 = "4.57"))
+  testthat::expect_identical(matrix_form(lsting)$strings[3, 1:2], c(AGE = "", BMRKR1 = "5.00"))
+})
+
+testthat::test_that("as_listing works with NA values in key cols", {
+  mtcars$gear[1:5] <- NA
```

Comment:
I think all is very good. Only I found a case that I think we should trim, when a row is only NAs. 
```r
data(mtcars)
mtcars[2, ]
lsting <- as_listing(
    mtcars,
    key_cols = c("gear", "carb"),
    disp_cols = "qsec"
)
```
produces something that ends like:
```
        8     14.6 
 NA     NA     NA  
```
I think the lastline is not needed